rotkehlchen/exchanges/binance.py

```diff
@@ -893,6 +916,92 @@ def query_online_trade_history(
 
         return trades
 
+    def _deserialize_fiat_movement(
+            self,
+            raw_data: Dict[str, Any],
+            transaction_type: Literal['deposit', 'withdrawal', 'buy', 'sell'],
+    ) -> Optional[NamedTuple]:
```

Comment:
The type here is wrong, and this is also why `make lint` fails.

Reply:
yeah, wanted to make the deserialize function compatible to the fiat/payments endpoint, how would i type-hint this correctly so that the linter wont complain? since it could also return an instance of Trade when deserializing results from fiat/payments?

Reply:
That should be `Optional[Union[Trade, AssetMovement]]`

Reply:
thx :)

Reply:
hm :/ linter is still complaining:

```
rotkehlchen/exchanges/binance.py:929: error: Incompatible return value type (got "AssetMovement", expected "Union[Trade, AssetAmount, None]")
Found 3 errors in 1 file (checked 459 source files)
rotkehlchen/exchanges/binance.py:1174: error: Incompatible types in assignment (expression has type "Union[Trade, AssetAmount, None]", variable has type "Optional[AssetMovement]")
```

Reply:
It's a bit more complicated than that. You would probably need to overload the types? Do you want me to do it to show you how? I can also squash all the merge/fix commits you have.

Reply:
for now i just made a seperate deserialize_fiat_payments function, so that linting passes.
but if you prefer to have them combined, please provide me with an function definition that passes the linter :)

Reply:
Thank you @dweissengruber! 

There is other examples in the codebase of how one uses mypy to overload function definitions. Also in their docs. I will just do it myself for now so I can show you how it works and you can know for next time. No worries.

(Assuming the problem is what I think it is :rofl: )

Reply:
Wait wait, so fiat payments come from a different endpoint from what I see.

And the function is not used in your PR at all. In which case different functions make sense. I will try to use it though. Will push on top of your PR.

Reply:
Added some more commits which should address everything. Tests still need to be written as I saw you removed them. I can take care of it a bit later. Will point you in the code where things needed changing.